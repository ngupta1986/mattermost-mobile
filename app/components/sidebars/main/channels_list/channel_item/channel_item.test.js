// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

import React from 'react';
import {shallow} from 'enzyme';

import Preferences from 'mattermost-redux/constants/preferences';

import ChannelItem from './channel_item.js';

jest.mock('react-intl');

describe('ChannelItem', () => {
    const baseProps = {
        channelId: 'channel_id',
        currentChannelId: 'current_channel_id',
        displayName: 'display_name',
        fake: false,
        isChannelMuted: false,
        isMyUser: true,
        isUnread: true,
        hasDraft: false,
        mentions: 0,
        navigator: {push: () => {}}, // eslint-disable-line no-empty-function
        onSelectChannel: () => {}, // eslint-disable-line no-empty-function
        shouldHideChannel: false,
        showUnreadForMsgs: true,
        status: 'online',
        teammateDeletedAt: 0,
        type: 'O',
        theme: Preferences.THEMES.default,
        unreadMsgs: 1,
        isArchived: false,
    };

    test('should match snapshot', () => {
        const wrapper = shallow(
            <ChannelItem {...baseProps}/>,
            {context: {intl: {formatMessage: jest.fn()}}},
        );

        expect(wrapper.getElement()).toMatchSnapshot();
    });

    test('should match snapshot for deactivated user', () => {
        const newProps = {
            ...baseProps,
            teammateDeletedAt: 100,
            type: 'D',
        };
        const wrapper = shallow(
            <ChannelItem {...newProps}/>,
            {context: {intl: {formatMessage: jest.fn()}}},
        );
        expect(wrapper.getElement()).toMatchSnapshot();
    });

    test('should match snapshot with draft', () => {
        const wrapper = shallow(
            <ChannelItem
                {...baseProps}
                hasDraft={true}
            />,
            {context: {intl: {formatMessage: jest.fn()}}},
        );

        expect(wrapper.getElement()).toMatchSnapshot();
    });
});
