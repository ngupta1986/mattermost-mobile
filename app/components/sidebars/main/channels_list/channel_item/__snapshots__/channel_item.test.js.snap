// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ChannelItem should match snapshot 1`] = `
<AnimatedComponent>
  <TouchableHighlight
    activeOpacity={0.85}
    delayPressOut={100}
    onLongPress={[Function]}
    onPress={[Function]}
    underlayColor="rgba(69,120,191,0.5)"
  >
    <Component
      style={
        Array [
          Object {
            "flex": 1,
            "flexDirection": "row",
            "height": 44,
          },
          undefined,
        ]
      }
    >
      <Component
        style={
          Array [
            Object {
              "alignItems": "center",
              "flex": 1,
              "flexDirection": "row",
              "paddingLeft": 16,
            },
            undefined,
          ]
        }
      >
        <ChannelIcon
          channelId="channel_id"
          hasDraft={false}
          isActive={false}
          isArchived={false}
          isInfo={false}
          isUnread={true}
          membersCount={1}
          size={16}
          status="online"
          teammateDeletedAt={0}
          theme={
            Object {
              "awayIndicator": "#ffbc42",
              "buttonBg": "#166de0",
              "buttonColor": "#ffffff",
              "centerChannelBg": "#ffffff",
              "centerChannelColor": "#3d3c40",
              "codeTheme": "github",
              "dndIndicator": "#f74343",
              "errorTextColor": "#fd5960",
              "linkColor": "#2389d7",
              "mentionBj": "#ffffff",
              "mentionColor": "#145dbf",
              "mentionHighlightBg": "#ffe577",
              "mentionHighlightLink": "#166de0",
              "newMessageSeparator": "#ff8800",
              "onlineIndicator": "#06d6a0",
              "sidebarBg": "#145dbf",
              "sidebarHeaderBg": "#1153ab",
              "sidebarHeaderTextColor": "#ffffff",
              "sidebarText": "#ffffff",
              "sidebarTextActiveBorder": "#579eff",
              "sidebarTextActiveColor": "#ffffff",
              "sidebarTextHoverBg": "#4578bf",
              "sidebarUnreadText": "#ffffff",
              "type": "Mattermost",
            }
          }
          type="O"
        />
        <Component
          ellipsizeMode="tail"
          numberOfLines={1}
          style={
            Array [
              Object {
                "color": "rgba(255,255,255,0.4)",
                "flex": 1,
                "fontSize": 14,
                "fontWeight": "600",
                "height": "100%",
                "lineHeight": 44,
                "paddingRight": 40,
                "textAlignVertical": "center",
              },
              Object {
                "color": "#ffffff",
              },
            ]
          }
        />
      </Component>
    </Component>
  </TouchableHighlight>
</AnimatedComponent>
`;

exports[`ChannelItem should match snapshot for deactivated user 1`] = `
<AnimatedComponent>
  <TouchableHighlight
    activeOpacity={0.85}
    delayPressOut={100}
    onLongPress={[Function]}
    onPress={[Function]}
    underlayColor="rgba(69,120,191,0.5)"
  >
    <Component
      style={
        Array [
          Object {
            "flex": 1,
            "flexDirection": "row",
            "height": 44,
          },
          undefined,
        ]
      }
    >
      <Component
        style={
          Array [
            Object {
              "alignItems": "center",
              "flex": 1,
              "flexDirection": "row",
              "paddingLeft": 16,
            },
            undefined,
          ]
        }
      >
        <ChannelIcon
          channelId="channel_id"
          hasDraft={false}
          isActive={false}
          isArchived={false}
          isInfo={false}
          isUnread={true}
          membersCount={1}
          size={16}
          status="online"
          teammateDeletedAt={100}
          theme={
            Object {
              "awayIndicator": "#ffbc42",
              "buttonBg": "#166de0",
              "buttonColor": "#ffffff",
              "centerChannelBg": "#ffffff",
              "centerChannelColor": "#3d3c40",
              "codeTheme": "github",
              "dndIndicator": "#f74343",
              "errorTextColor": "#fd5960",
              "linkColor": "#2389d7",
              "mentionBj": "#ffffff",
              "mentionColor": "#145dbf",
              "mentionHighlightBg": "#ffe577",
              "mentionHighlightLink": "#166de0",
              "newMessageSeparator": "#ff8800",
              "onlineIndicator": "#06d6a0",
              "sidebarBg": "#145dbf",
              "sidebarHeaderBg": "#1153ab",
              "sidebarHeaderTextColor": "#ffffff",
              "sidebarText": "#ffffff",
              "sidebarTextActiveBorder": "#579eff",
              "sidebarTextActiveColor": "#ffffff",
              "sidebarTextHoverBg": "#4578bf",
              "sidebarUnreadText": "#ffffff",
              "type": "Mattermost",
            }
          }
          type="D"
        />
        <Component
          ellipsizeMode="tail"
          numberOfLines={1}
          style={
            Array [
              Object {
                "color": "rgba(255,255,255,0.4)",
                "flex": 1,
                "fontSize": 14,
                "fontWeight": "600",
                "height": "100%",
                "lineHeight": 44,
                "paddingRight": 40,
                "textAlignVertical": "center",
              },
              Object {
                "color": "#ffffff",
              },
            ]
          }
        />
      </Component>
    </Component>
  </TouchableHighlight>
</AnimatedComponent>
`;

exports[`ChannelItem should match snapshot with draft 1`] = `
<AnimatedComponent>
  <TouchableHighlight
    activeOpacity={0.85}
    delayPressOut={100}
    onLongPress={[Function]}
    onPress={[Function]}
    underlayColor="rgba(69,120,191,0.5)"
  >
    <Component
      style={
        Array [
          Object {
            "flex": 1,
            "flexDirection": "row",
            "height": 44,
          },
          undefined,
        ]
      }
    >
      <Component
        style={
          Array [
            Object {
              "alignItems": "center",
              "flex": 1,
              "flexDirection": "row",
              "paddingLeft": 16,
            },
            undefined,
          ]
        }
      >
        <ChannelIcon
          channelId="channel_id"
          hasDraft={true}
          isActive={false}
          isArchived={false}
          isInfo={false}
          isUnread={true}
          membersCount={1}
          size={16}
          status="online"
          teammateDeletedAt={0}
          theme={
            Object {
              "awayIndicator": "#ffbc42",
              "buttonBg": "#166de0",
              "buttonColor": "#ffffff",
              "centerChannelBg": "#ffffff",
              "centerChannelColor": "#3d3c40",
              "codeTheme": "github",
              "dndIndicator": "#f74343",
              "errorTextColor": "#fd5960",
              "linkColor": "#2389d7",
              "mentionBj": "#ffffff",
              "mentionColor": "#145dbf",
              "mentionHighlightBg": "#ffe577",
              "mentionHighlightLink": "#166de0",
              "newMessageSeparator": "#ff8800",
              "onlineIndicator": "#06d6a0",
              "sidebarBg": "#145dbf",
              "sidebarHeaderBg": "#1153ab",
              "sidebarHeaderTextColor": "#ffffff",
              "sidebarText": "#ffffff",
              "sidebarTextActiveBorder": "#579eff",
              "sidebarTextActiveColor": "#ffffff",
              "sidebarTextHoverBg": "#4578bf",
              "sidebarUnreadText": "#ffffff",
              "type": "Mattermost",
            }
          }
          type="O"
        />
        <Component
          ellipsizeMode="tail"
          numberOfLines={1}
          style={
            Array [
              Object {
                "color": "rgba(255,255,255,0.4)",
                "flex": 1,
                "fontSize": 14,
                "fontWeight": "600",
                "height": "100%",
                "lineHeight": 44,
                "paddingRight": 40,
                "textAlignVertical": "center",
              },
              Object {
                "color": "#ffffff",
              },
            ]
          }
        />
      </Component>
    </Component>
  </TouchableHighlight>
</AnimatedComponent>
`;
